DOC-614 | Added `readOwnWrites` option for UPSERT operations (#409)
arangodb/docs-hugo: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md

Read more about [`UPSERT` operations](../../aql/high-level-operations/upsert.md) in AQL.

### `readOwnWrites` option for `UPSERT` operations

> The `readOwnWrites` option has been added for `UPSERT` operations. The default

Reviewer: ```suggestion
A `readOwnWrites` option has been added for `UPSERT` operations. The default
```

> value is `true`. When set to `true`, an `UPSERT` operation processes its

Reviewer: ```suggestion
value is `true` and the behavior is identical to previous versions of ArangoDB that
do not have this option. When enabled, an `UPSERT` operation processes its
```

inputs one by one. This way, the operation can observe its own writes and can
handle modifying the same target document multiple times in the same query.

When the option is set to `false`, an `UPSERT` operation processes its inputs

> in batches. A batch has normally 1000 inputs, so it can lead to a faster execution.

Reviewer: ```suggestion
in batches. Normally, a batch has 1000 inputs, which can lead to a faster execution.
```

However, when using batches, the `UPSERT` operation cannot observe its own writes.

> Therefore, the `readOwnWrites` option should be set to `false` only if it can
> be guaranteed that the input of the `UPSERT` leads to disjoint documents being

Reviewer: ```suggestion
Therefore, you should only set the `readOwnWrites` option to `false` if you can
guarantee that the input of the `UPSERT` leads to disjoint documents being
```
